core/container/intrusive/list/doc.odin

```diff
@@ -0,0 +1,39 @@
+/*
+Package list implements an intrusive doubly-linked list.
+
+An intrusive container requires a `Node` to be embedded in your own structure, like this:
+
+	My_String :: struct {
+		node:  list.Node,
```

Comment:
A minor point, but it can be helpful to know that `using node: list.Node,` is also allowed thanks to the `container_of` magic. A more naïve implementation of an intrusive list may have not worked with the `Node`'s fields being part of the container instead of its child, if that makes sense.